### PR TITLE
Make TileRenderer smoothing configurable

### PR DESCRIPTION
**Pull request by philippjfr**
As the title says, exposes control over canvas image smoothing on the TileRenderer.

- [x] issues: fixes https://github.com/bokeh/bokeh/issues/8172
- [x] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by philippjfr** [comment]
Not sure how I can test this, since the smoothing is only disabled temporarily.

File: bokeh/models/renderers.py [hunk 1]
```diff
@@ -53,6 +53,10 @@ class TileRenderer(DataRenderer):
     tile opacity 0.0 - 1.0
     """)
 
+    smoothing = Bool(default=True, help="""
+    Enable image smoothing for the rendered tiles
```
Comment (mattpap) [inline bokeh/models/renderers.py]: A trailing period would be nice.

**Comment by mattpap** [comment]
> Not sure how I can test this, since the smoothing is only disabled temporarily.

I would just create a static integration example with the same tile source side by side with smoothing enabled and disabled. If something get broken, we will see thin in image diff.

**Comment by bryevdv** [comment]
@philippjfr does the suggested test sound reasonable for now? Do you have any questions about implementing it?

**Comment by philippjfr** [comment]
>does the suggested test sound reasonable for now? Do you have any questions about implementing it?

Yes, had half forgotten about this after I addressed the flicker issue. If you could point me to where the example would live I'll add it tonight. Not sure how noticeable the difference between smoothed and non-smoothed will be but presumably an image diff will definitely pick up on it.

**Comment by philippjfr** [comment]
Let me know if the example looks okay.

**Comment by philippjfr** [comment]
Ah just realized I put it in tests/integration and you wanted it in examples/integration, moving it now.

**Comment by mattpap** [comment]
Looks good. Difference in smoothness is clearly visible.

**Pull request merged**